### PR TITLE
Add RowTable and ColTable

### LINKED ISSUE #62: Introduce RowTable and ColTable transforms
For row table inputs, will the output of a transform necessarily have `rowaccess`?  Is there a general principle guiding the form of the output in general? Can I arrange that the ouput is `Tables.materialize`d? 

In MLJ  we generally `materialize` based on input type, but is has been argued that an option to specify the output type would be useful. I'm wondering what the authors of  TableTransforms.jl think about such questions.

### PR DESCRIPTION
**Pull request by ceferisbarov**
Fix #62 

Added RowTable transform and its tests.

### CODE REVIEW

File: src/transforms/rowtable.jl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+"""
+    RowTable()
+
+The transform that applies the function `Tables.rowtable` to to the input table.
+
+# Examples
+
+```julia
+RowTable()
+```
```
Comment (juliohm) [inline src/transforms/rowtable.jl:14]:
```suggestion
```

File: src/transforms/rowtable.jl [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+"""
+struct RowTable <: Transform end
```
Comment (juliohm) [inline src/transforms/rowtable.jl:16]: This transform should be Stateless

File: test/transforms.jl [hunk 3]
```diff
@@ -859,6 +859,19 @@
...
+  @testset "RowTable" begin
+    a = [3, 2, 1, 4, 5, 3]
+    b = [1, 4, 4, 5, 8, 5]
+    c = [1, 1, 6, 2, 4, 1]
+    t = Table(; a, b, c)
+    T = RowTable()
+    n, c = apply(T, t)
+    tₒ = revert(T, n, c)
+    @test typeof(t) <: Table
+    @test typeof(n) <: Vector
```
Comment (juliohm) [inline test/transforms.jl:871]: Should also check that the new table is Tables.rowaccess

**Review by juliohm — CHANGES_REQUESTED** [review]
Added a first round of reviews. Feel free to add ColTable to this PR as well given that it is pretty much the same code.

File: test/transforms.jl (hunk 3 above)
Comment (eliascarv) [reply test/transforms.jl:871]: I think this check is not necessary:
```julia
@test typeof(t) <: Table
```

File: src/transforms/coltable.jl [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+"""
+    ColTable()
+
+The transform that applies the function `Tables.columntable` to to the input table.
+"""
+struct ColTable <: Stateless end
+
+apply(::ColTable, table) = Tables.columntable(table), table
+
+revert(::ColTable, table, cache) = cache
```
Comment (eliascarv) [inline src/transforms/coltable.jl:14]:
```suggestion
revert(::ColTable, newtable, cache) = cache
```

File: src/transforms/rowtable.jl [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+"""
+    RowTable()
+
+The transform that applies the function `Tables.rowtable` to to the input table.
+"""
+struct RowTable <: Stateless end
+
+apply(::RowTable, table) = Tables.rowtable(table), table
+
+revert(::RowTable, table, cache) = cache
```
Comment (eliascarv) [inline src/transforms/rowtable.jl:14]:
```suggestion
revert(::RowTable, newtable, cache) = cache
```

File: src/transforms/coltable.jl (hunk 4 above)
Comment (juliohm) [reply src/transforms/coltable.jl:14]: @ceferisbarov please pay more attention to the code style. Try to copy/paste the variable names that are used consistently across the transforms. Thank you @eliascarv for spotting that.

**Review by eliascarv — APPROVED** [review]

**Review by juliohm — CHANGES_REQUESTED** [review]
Awesome, now we need to add the two transforms to the documentation like the other transforms.

File: test/transforms.jl [hunk 6]
```diff
@@ -859,6 +859,32 @@
...
+  @testset "RowTable" begin
+    a = [3, 2, 1, 4, 5, 3]
+    b = [1, 4, 4, 5, 8, 5]
+    c = [1, 1, 6, 2, 4, 1]
+    t = Table(; a, b, c)
+    T = RowTable()
+    n, c = apply(T, t)
+    tₒ = revert(T, n, c)
+    @test typeof(n) <: Vector
+    @test Tables.rowaccess(typeof(n))
```
Comment (juliohm) [inline test/transforms.jl:871]: Do we really need the typeof here @ceferisbarov ?

File: test/transforms.jl [hunk 7]
```diff
@@ -859,6 +859,32 @@
...
+  @testset "ColTable" begin
+    a = [3, 2, 1, 4, 5, 3]
+    b = [1, 4, 4, 5, 8, 5]
+    c = [1, 1, 6, 2, 4, 1]
+    t = Table(; a, b, c)
+    T = ColTable()
+    n, c = apply(T, t)
+    tₒ = revert(T, n, c)
+    @test typeof(n) <: NamedTuple
+    @test Tables.columnaccess(typeof(n))
```
Comment (juliohm) [inline test/transforms.jl:884]: Same here, do we need typeof?

File: test/transforms.jl (hunk 6 above)
Comment (ceferisbarov) [reply test/transforms.jl:871]: From Tables.jl API:  
_

Tables.rowaccess(table)    Declare that your table type defines a Tables.rows(table) method_  
  
This seems to be the intended use, but it also works when I remove `typeof`.

**Comment by juliohm** [comment]
Awesome! Thank you! 💯

**Pull request merged**